cot: add page (#8205)
tldr-pages/tldr: pages/osx/cot.md

# cot

> The Plain-Text Editor for macOS.
> More information: <https://coteditor.com/>.

- Start CotEditor:

`cot`

> - Open a new blank document:
> 
> `cot --new`
> 
> - Open specific files:
> 
> `cot {{path/to/file1 path/to/file2 ...}}`

Reviewer: ```suggestion
- Open specific files in background:

`cot {{path/to/file1 path/to/file2 ...}}`

- Open a new blank document in background:

`cot --new`
```

- See #8018 why I suggested this. 😄
- Is it right that by default everything is opened in background in this editor? If it's true it's better to specify this explicitly assuming that in general editor opens files in foreground by default (until otherwise noted explicitly).

I can't try this editor, so please check my edits carefully.

Author: After testing the behavior of running `cot`, it appears that the windows will in fact always open in the background, even though the `-g`/`--background` flag is not provided. This might be a bug with `cot`, although I'm not completely sure.

Because of this, I'd prefer to not specifically state that the command opens file in the background, although I can see why we might want to document it anyway, as it is assumed that `cot` would open files in the foreground by default.

(I will make the change to swap the order of `cot --new` and `cot {{path/to/file1 path/to/file2 ...}}` right now)

Author: Also, to me, "Open specific files in **the** background" sounds a bit better than "Open specific files in background". If we keep the "in background" wording (see comment above), would that be a better way to word it, or should I keep it without the "the"?

Reviewer: Let's add `the`.

Author: Okay, cool. I've made these changes.

> - Open and wait until done editing a specific file:

Reviewer 2: ```suggestion
- Open and wait until finishing editing a specific file:
```

Author: Here's my reply from [the other PR](https://github.com/tldr-pages/tldr/pull/8206) for reference:
> This exact line was taken directly from [the `webstorm` tldr-page](https://github.com/tldr-pages/tldr/blob/da55f621eb9c77d579b4353d78cc4c9bec39c4f3/pages/common/webstorm.md), but I agree that "finishing" is better than "done".
> 
> I think "finished" might be a better form of the word than "finishing" to use here, though.

> - Open and wait until finished editing a specific file:

Reviewer 3: ```suggestion
- Open a specific file in the foreground (blocking the terminal until the editor is closed):
```

Author: This line is worded the same way that #8206 was worded. In addition to making this change, should I also open a PR to update the wording for `mate`?

> - Open a specific file (blocking the terminal until the editor is closed):

Reviewer: ```suggestion
- Open a specific file in the foreground:
```

Reviewer 3: I would be fine with that, but I just have the concern that it's not really clear, what `in the foreground` means here.

Reviewer: `in the background` expression also can be confusing. :) Maybe it's better to write what `in the background` actually means such as `without blocking a terminal until the editor is closed`.

Reviewer 3: That's why I went for this solution, which seemed most natural to me. However, that's just one opinion. I'd say in cases like this we should let the author decide (@superatomic).

Author: I was interpreting foreground/background to be whether the window that was opened was sent to the foreground, not whether the terminal blocks.

As it currently stands, either interpretation is correct, although it's probably better to use more descriptive language to avoid confusion.

For this line, I personally like "Open and wait until finished editing a specific file", which was the original text before it was changed. I'd suppose "Open a specific file and wait until it is closed" might be a good way to phrase this as well.

Reviewer 3: I like the last suggestion. But `wait` is a bit ambiguous. How about `Open a specific file and block the terminal until it is closed`?

Author: That sounds good. I chose the word "wait" because it was the name of the option, but I like yours more.

> - Open a specific file in the foreground (blocking the terminal until the editor is closed):
> 
> `cot --wait {{path/to/file}}`

Reviewer: ```suggestion
- Open specific files:

`cot --wait {{path/to/file1 path/to/file2 ...}}`
```

Author: Should this still keep some information about how this command blocks the terminal until the editor is closed? Maybe "Open specific files (blocking the terminal until the editor is closed)"?

Reviewer: I don't think so. Anyway we have to decide whether to write such info everywhere where appropriate or not. 😄

Reviewer 3: I think we should keep the previous version. We somehow have to explain what the `--wait` option does. Otherwise this example would be useless.

> - Open a file with the cursor at a specific line and column:

Reviewer 3: ```suggestion
- Open a specific file with the cursor at a specific line and column:
```

Author: Same deal here. I can make a new PR to give `mate` the same wording as `cot` once this PR is merged.
